### PR TITLE
Add WeightSize to BatchNorm and Conv Layer.

### PR DESCRIPTION
**Pull request by kartikdutt18**
Hey everyone,
This PR adds weights size to BatchNorm and Conv Layer.
Related Issue : #2590

Regards.

### CODE REVIEW

**Comment by UtR491** [comment]
I was wondering, should we also include a test that checks the correctness of the visitor for some layer that has both the Parameters() and Model() functions? Maybe the GRU layer? Since it is not yet merged, I can include it in my PR or you can include it in this one.

**Comment by kartikdutt18** [comment]
Since this PR only deals with BatchNorm and Conv layer, I would consider it out of the scope of this PR. Feel free to open an issue or PR for the same. Better test coverage is always good. Thanks.

File: src/mlpack/methods/ann/layer/convolution.hpp [hunk 1]
```diff
@@ -253,6 +253,12 @@ class Convolution
   //! Modify the right padding width.
   size_t& PadWRight() { return padWRight; }
 
+  //! Get size of weights for the layer.
+  size_t WeightSize() const
+  {
+    return (outSize * inSize * kernelWidth *kernelHeight) + outSize;
```
Comment (rcurtin) [inline src/mlpack/methods/ann/layer/convolution.hpp:259]:
```suggestion
    return (outSize * inSize * kernelWidth * kernelHeight) + outSize;
```
:)

File: src/mlpack/tests/ann_visitor_test.cpp [hunk 2]
```diff
@@ -53,6 +53,41 @@ TEST_CASE("BiasSetVisitorTest", "[ANNVisitorTest]")
...
+  size_t randomKernelHeight = arma::randi(arma::distr_param(1, 100));
+
+  LayerTypes<> convLayer = new Convolution<>(randomInSize, randomOutSize,
+      randomKernelWidth, randomKernelHeight);
+
+  size_t weightSize = boost::apply_visitor(WeightSizeVisitor(),
+                                           convLayer);
+
+  REQUIRE(weightSize == (randomOutSize * randomInSize * randomKernelWidth *
+      randomKernelHeight) + randomOutSize);
```
Comment (rcurtin) [inline src/mlpack/tests/ann_visitor_test.cpp:73]: Might be better to test against `convLayer.Parameters().n_elem`?

**Review by rcurtin — COMMENTED** [review]
Hey @kartikdutt18, looking good---just a couple comments. :+1:

File: src/mlpack/tests/ann_visitor_test.cpp (hunk 2 above)
Comment (kartikdutt18) [reply src/mlpack/tests/ann_visitor_test.cpp:73]: Yes that would be better. Made the changes to reflect the same. Thanks for the review.

**Review by rcurtin — APPROVED** [review]
Awesome, thanks @kartikdutt18! :+1:

**Comment by kartikdutt18** [comment]
Thanks a lot @rcurtin for the review. Bringing it in.

**Pull request merged**